_posts/2016-09-22-approaching-tdd-outside-android-ii.md

```diff
@@ -0,0 +1,113 @@
+
+<ul style="display: inline-block; list-style: none; text-align: left;">
+<li><span style="color: #d32f2f; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/8520fbba9b3ed7e4b330544c23eb1027db77f897">8520fb</a> - The test has to verify that when the activity is launched, the view gets attached to the account and the BankAccount <i>showStatement</i> method is called. As mentioned before, we do not have control over the activity instantiation so we need to use the ServiceLocator to provide a test BankAccount for the activity.</li>
+<li><span style="color: #388e3c; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/fcb4030f88150ac718f95a3ebb1740b24c55c9af">fcb403</a> - To make the test pass we need to obtain the account reference from the ServiceLocator, attach the ShowStatementView (activity) to the account instance, and call its showStatement method.</li>
+<li><span style="color: #1976d2; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/bd961ceaf73948806218999cd1f301c417ebcf2c">bd961c</a> - We added all the code inside the onCreate method. Following the same reasoning as in the previous refactor, we extract a <i>initAccount</i> method that wraps bank account initialisation code.</li></ul>
+
+The activity is now complete. The next step would be to implement the next collaborator, the <b>ServiceLocator</b>. The ServiceLocator job is to provide BankAccount instances to its clients. Let's implement it doing a second iteration of the inner loop cycle.
+
+<ul style="display: inline-block; list-style: none; text-align: left;">
+<li><span style="color: #d32f2f; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/7fb17baff0f7ad1400560cb841dfdcdf78e66177">7fb17b</a> - The test for the ServiceLocator has to assert that the provided BankAccount is not null.</li>
+<li><span style="color: #388e3c; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/8ce1466630124fb680dae0d72b421c429d929843">8ce146</a> - In this case is just as simple as returning a new BankAccunt instance.</li>
+<li><span style="color: #1976d2; padding-right: 5px;">●</span>There is no refactor to do in this step, so let's move to the next iteration.</li></ul>
+
+Now that the outer layer is done, it is time to move to the next level of abstraction and implement the <b>BankAccount</b>. Among the BankAccount public methods we need to decide which to implement first - <i>deposit()</i>, <i>withdraw()</i> and <i>showStatement()</i>. We recommend following the order in which the methods are used in the acceptance test. We will show how we can ensure the order later on. In this case, the method that is used first is <b>deposit()</b>. Let's implement it by going through another iteration of the inner loop cycle.
```

Comment:
@sandromancuso We used the term `BankAccount` everyone, even in the provided code sample. I think it's better to leave it as it is, otherwise, it may be confusing to the reader to find both BankAccount and AccountServices across the posts and source code